uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt5/client/paho/MqttConnectionImpl.java

```diff
@@ -150,15 +150,22 @@ public MqttPublishReply publish(long timeout, @NonNull Message message) {
         mqttMessage.setPayload(message.getPayload());
         mqttMessage.setRetained(message.isRetain());
 
+        MqttProperties properties = new MqttProperties();
         if (message.getUserProperties() != null && !message.getUserProperties().isEmpty()) {
-            MqttProperties properties = new MqttProperties();
             List<UserProperty> userProperties = convertToUserProperties(message.getUserProperties());
             properties.setUserProperties(userProperties);
             userProperties.forEach(p -> logger.atInfo()
                     .log("Publish MQTT userProperties: {}, {}", p.getKey(), p.getValue()));
-            mqttMessage.setProperties(properties);
         }
 
+        String contentType = message.getContentType();
+        if (contentType != null && !contentType.isEmpty()) {
```

Comment:
Empty content type is valid value and should be passed.
So remove isEmpty() check.